src/networking.c

```diff
@@ -2634,13 +2634,28 @@ NULL
             addReplyBulk(c,c->name);
         else
             addReplyNull(c);
-    } else if (!strcasecmp(c->argv[1]->ptr,"pause") && c->argc == 3) {
-        /* CLIENT PAUSE */
+    } else if (!strcasecmp(c->argv[1]->ptr,"unpause") && c->argc == 2) {
+        /* CLIENT UNPAUSE */
+        unpauseClients(1);
+        addReply(c,shared.ok);
+    } else if (!strcasecmp(c->argv[1]->ptr,"pause") && (c->argc == 3 ||
+                                                        c->argc == 4))
+    {
+        /* CLIENT PAUSE TIMEOUT [WRITE] */
         long long duration;
+        int type = CLIENT_PAUSE_ALL;
+        if (c->argc == 4) {
+            if (!strcasecmp(c->argv[3]->ptr,"write")) {
```

Comment:
it just occurred to me that maybe it would be nice to also have a READ pause feature.
i.e. one that will still allow PING, INFO, CLIENT, SHUTDOWN, but will refuse GET, DBSIZE and any access to the keyspace.
what do you think? (should be very easy to implement now)

Reply:
I'm not convinced it's useful. If there is a usecase you think makes sense, I agree it'll be straightforward to implement.

Reply:
one usecase would have been during loading, and busy scripts.
instead of responding with error to all requests, we can just postpone them.
this can make the life of users much easier (kinda like the annoying EAGAIN retry that's needed just because the process got and handled a signal).
but i agree we can leave this for a later stage.